### PR TITLE
solution

### PR DESCRIPTION
**Pull request by mariiahorbova**

### CODE REVIEW

**Review by erikagayan — APPROVED** [review]

File: app/create_file.py [hunk 1]
```diff
@@ -1 +1,52 @@
...
+                file.write(line + "\n")
+
+    print(f"File '{filename}' created successfully at '{file_path}'.")
+
+
+def process_arguments(args: str) -> None:
+    if "-d" in args:
+        directory_index = args.index("-d") + 1
+        directory_path = os.path.join(*args[directory_index:])
+        os.makedirs(directory_path, exist_ok=True)
```
Comment (Oleg-s-s) [inline app/create_file.py:30]: image
this part work wrong

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 2]
```diff
@@ -1 +1,58 @@
...
+
+def process_arguments(args: List[str]) -> None:
+    if "-d" in args:
+        directory_index = args.index("-d") + 1
+        path = [*args[directory_index:]]
+        if "-f" in path:
+            p_index = path.index("-f")
+            path = path[:p_index]
+            directory_path = os.path.join(*path)
+        else:
```
Comment (LiudmylaKulinchenko) [inline app/create_file.py:34]: try to rewrirte without else here and in 37 line. Your code will be more clear and shorter

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 3]
```diff
@@ -1 +1,55 @@
...
+def process_arguments(args: List[str]) -> None:
+    directory_path = "."
+    if "-d" in args:
+        directory_index = args.index("-d") + 1
+        path = [*args[directory_index:]]
+        if "-f" in path:
+            p_index = path.index("-f")
+            path = path[:p_index]
+        directory_path = os.path.join(*path)
+        os.makedirs(directory_path, exist_ok=True)
```
Comment (YuriiKindrat) [inline app/create_file.py:35]: will this work if we'll swap the flags
f.e. `-f file.txt -d dir1/dir2`
hint: try to use argparse here, to optimize ur code

File: app/create_file.py [hunk 4]
```diff
@@ -1 +1,55 @@
...
+        with open(file_path, "a") as file:
+            file.write("\n\n")
+            file.write(timestamp + "\n")
+            for line in content:
+                file.write(line + "\n")
+    else:
+        with open(file_path, "w") as file:
+            file.write(timestamp + "\n")
+            for line in content:
+                file.write(line + "\n")
```
Comment (YuriiKindrat) [inline app/create_file.py:21]: u can use `a` mode for both cases
also make ur code DRY

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

File: app/create_file.py (hunk 3 above)
Comment (mariiahorbova) [reply app/create_file.py:35]: image
it works but i get why i need to change

**Review by LiudmylaKulinchenko — APPROVED** [review]
Well done 😸

**Pull request closed without merging**